### DISCUSSION TITLE
Plans going forward with RPM based distros / state of CentOS

### DISCUSSION THREAD

**Discussion by janaurka** [gruntwork-io/knowledge-base#286] [Help] [answered]
Hey there

Since [CentOS Stream](https://www.centos.org/centos-stream/) essentially made a big part of the "industry", which was leveraging CentOS for whatever purpose, move to [Rocky Linux](https://rockylinux.org/) and/or [AlmaLinux](https://almalinux.org/), to have a similar experience to running actually RHEL and not having a rolling release which might introduce new fun bugs, what is Gruntworks plan regarding which Distro/OS should be supported in the future?

Right now, when for example looking at the `bash-commons` [`os.sh`](https://github.com/gruntwork-io/bash-commons/blob/master/modules/bash-commons/src/os.sh) which does the "which os/distro are you using" part for lot’s of stuff around Gruntworks, it only support CentOS and RHEL. And then for example the [`ntp-module`](https://github.com/gruntwork-io/terraform-aws-security/blob/master/modules/ntp/install.sh#L52-L69) limits this support to CentOS, what is the roadmap for supported OS/Distros on the RPM side of life in the future? And is there a timeline for getting this implemented? Since CentOS is now pretty much rolling, supporting it might become ... interesting. 

From my side, I guess it would make sense to at least have 100% support for RHEL and at least 1 exact copy of RHEL, which apparently at the moment `rocky` and `alma` are the frontrunners for.

**Comment by gcagle3** [comment, marked as answer]
Thanks for relaying your interest in support for alternative Linux distros. Currently AWS does list Rocky Linux in its[ supported OSes](https://docs.aws.amazon.com/systems-manager/latest/userguide/prereqs-operating-systems.html#prereqs-os-linux), and others may become increasingly popular. For now we intend to stick to the OSes that we currently support and will be looking to gradually add more support as more users like you express interest. Being selective about what new features to support enables Gruntwork to deliver the best quality of modules and support. But as we grow we'll use this input to inform our next priorities.
